### DISCUSSION TITLE
Changing brand or still wait?

### DISCUSSION THREAD

**Discussion by fmarzocca** [JurajNyiri/HomeAssistant-Tapo-Control#643] [Discuss]
I was using this library in a script executing from Node Red to switch Privacy Mode ON/OFF and to read camera State.

As everybody knows, this library is no more working for months, due to a change in Tapo authentication process.
Disconnecting the camera from internet is not a solution, as I wouldn't be able to watch it from remote (App dosen't work from VPN) and I will not receive any notification message.

My question is: is it worth to wait for Tapo disclosing their new procedure or should I seriously start looking for other brands?

P.S.: I can use VLC on my smartphone and VPN to watch the cameras from remote, but I won't receive any notification anyway.)

**Comment by JurajNyiri** [comment]
You can also downgrade firmware, or use HA to access the streams and notifications, or other technologies like NVR, frigate etc for notifications as well.

**Reply by fmarzocca** [reply]
If I downgrade the firmware, I still have to block internet otherwise it will upgrade again. There is no "Block firmware" in Tapo App.

**Reply by JurajNyiri** [reply]
That is incorrect, you can disable firmware auto upgrade, even through the integration.

**Reply by fmarzocca** [reply]
> use HA to access the streams and notifications, or other technologies like NVR, frigate etc for notifications as well.

should I have to install HA or Frigate just tp watch at 2 small cameras? That is not worth for sure.

**Reply by fmarzocca** [reply]
> That is incorrect, you can disable firmware auto upgrade, even through the integration.

As I said, I am not using HA. Only NodeRed with pytapo

**Reply by JurajNyiri** [reply]
You can disable it via app or pytapo.

**Reply by fmarzocca** [reply]
> You can disable it via app or pytapo.

I can't find it in App. And there are no instructions for pytapo for this.

**Reply by JurajNyiri** [reply]
Camera - settings - firmware update - auto update. 

If you are using pytapo, you can read or write source code, look through available function definitions and/or search for firmware, its [here](https://github.com/JurajNyiri/pytapo/blob/main/pytapo/__init__.py#L923).

**Reply by fmarzocca** [reply]
> You can disable it via app or pytapo.

C200 v1 do not have this capability, neither through App nor through pytapo. 
I have no way to restore my 2 C200, that's why I am changing brand

**Reply by JurajNyiri** [reply]
Incorrect. C200 V1 has firmware available all the way since 2021 many of which does not auto upgrade. 

It only auto upgrades if the option is there.

**Reply by fmarzocca** [reply]
> Incorrect. C200 V1 has firmware available all the way since 2021 many of which does not auto upgrade.
> 
> It only auto upgrades if the option is there.

I have 2 C200 cameras, and they do NOT have this option. So, if I connect the cameras to internet, they always update the firmware. No way to block this from App or pytapo.

**Reply by JurajNyiri** [reply]
In that case they do not auto upgrade as specified above.

**Reply by fmarzocca** [reply]
> In that case they do not auto upgrade as specified above.

Wrong. It happened to me.

**Reply by JurajNyiri** [reply]
🤷 I have 5x c200 all over firmwares. What I said is correct. The option documented works for all the users tried it. 

I am trying to help but you are just being negative, you already said you threw them in trash. I am open source dev and have absolutely no benefit from you doing whatever you want with the camera, do as you wish.

**Reply by fmarzocca** [reply]
> 
> I am trying to help but you are just being negative, you already said you threw them in trash. I am open source dev and have absolutely no benefit from you doing whatever you want with the camera, do as you wish.

Oh, if I have offended you in any way, I truly apologize! I am not blaming you, but Tapo!!
I appreciate your hard work and I am following your efforts for months, while I am using pytapo for years! I am an OS developer too, so I truly understand your position. I am sorry. again. I'm just stressed because these cameras are wasting a lot of my time.

**Reply by fmarzocca** [reply]
If I manage to downgrade to 1.3.6 and then connect C200v1 to internet, pytapo authentication will still be working?

**Reply by JurajNyiri** [reply]
It needs to be build 230921 and lower. Yes, then it will work as stated in the [docs](https://github.com/JurajNyiri/HomeAssistant-Tapo-Control?tab=readme-ov-file#integration-does-not-work-temporarily-on-firmware-build-230921-and-higher).

**Reply by fmarzocca** [reply]
> docs

Yes, it is now 230424. Thanks

**Comment by JurajNyiri** [comment]
>  is it worth to wait for Tapo disclosing their new procedure or should I seriously start looking for other brands?

The last update I have from them is from May, that the endpoint will be most likely ready by end of June. Since then, I have reached out twice with no response, last time on 2024-07-16.

**Reply by fmarzocca** [reply]
Thanks
